pymeasure/instruments/scientificmagnetics/scimag.py

```diff
@@ -0,0 +1,448 @@
+        values={False: 0, True: 1},
+        map_values=True
+    )
+
+    ramp_target = Instrument.setting(
+        "R%d",
+        """Sets Ramp Target. Can take the values "Z"ero, "L"ower, "U"pper.
+        """,
+        validator=strict_discrete_set,
+        values={"Z": 0, "L": 1, "U": 2},
+        map_values=True
+    )
+
+    persistent_mode_heater = Instrument.setting(
+        "H%d",
+        """Sets the persistent mode.
+        Can take the values "O"ff, "C"onditional on, "U"nconditional on, "R"eset persistent current mode to zero.
+        WARNING: Unconditional on will turn on the heater without any interlocks! Use at own risk.
+        Standard mode to switch heater on is Conditional.
+        Simplified Commands are "True" (Conditional on) and "False" (Off).
+        """,
+        validator=strict_discrete_set,
+        values={"O": 0, "C": 1, "U": 2, "R": 9, False: 0, True: 1},
+        map_values=True
+    )
+
+    ramp_rate = Instrument.setting(
+        "A%08.5f",
+        """Sets the Ramp rate in A/s.""",
+        validator=strict_range,
+        values=[0, _RAMP_RATE_MAX],
+    )
+
+    external_trip = Instrument.setting(
+        "X%d",
+        """Sets the External trip:
+         False = Off, True = Simple trip On, "Auto" = Auto-rampdown On.""",
+        validator=strict_discrete_set,
+        values={False: 0, True: 1, "Auto": 4},
+        map_values=True
+    )
+
+    terminal_voltage_limit = Instrument.setting(
+        "Y%04.1f",
+        """Sets the terminal voltage limit to nn.n V.""",
+        validator=strict_range,
+        values=[0, _VOLTAGE_LIMIT_MAX]
+    )
+
+    heater_current_output = Instrument.setting(
+        "W%03d",
+        """Sets the Heater current output to nnn mA.""",
+        validator=strict_range,
+        values=[0, 255]
+    )
+
+    calibrate_tesla_per_amp = Instrument.setting(
+        "C%08.6f",
+        """Calibrate the output to Tesla per Amp value of the superconducting magnet.
+         The Value is assumed to lie between 0.01 and 0.5 T/A
+         and will be set to zero if a number outside this range is entered.
+         """,
+        validator=strict_range,
+        values=[0.01, 0.5],
+    )
+
+    lower_setpoint = Instrument.setting(
+        "L%08.4f",
+        """Sets the lower setpoint. 
+        Unit depends on the unit set in the magnet controller.
+        Unit can be "T" or "A" and must be set before.
+        """,
+        validator=strict_range,
+        values=[0, 200]
+    )
+
+    upper_setpoint = Instrument.setting(
+        "U%08.4f",
+        """Sets the upper setpoint. 
+        Unit depends on the unit set in the magnet controller.
+        Unit can be "T" or "A" and must be set before.
+        """,
+        validator=strict_range,
+        values=[0, 200]
+    )
+
+    output_parameters_amps = Instrument.measurement(
+        "G",
+        """Returns dict of output parameters.""",
+        preprocess_reply=extract_value_output_parameters_amps
```

Comment:
See my remark in `instruments.py`; I think you can better pass it as a `get_process`.